src/main/java/tickbot/storage/DataStorage.java

```diff
@@ -0,0 +1,113 @@
+package tickbot.storage;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import com.opencsv.CSVReader;
+import com.opencsv.CSVWriter;
+
+import net.harawata.appdirs.AppDirs;
+import net.harawata.appdirs.AppDirsFactory;
+import tickbot.exception.CorruptedDataException;
+import tickbot.task.Deadline;
+import tickbot.task.Event;
+import tickbot.task.Task;
+import tickbot.task.Todo;
+
+/**
+ * The class to represent the data storage.
+ */
+public class DataStorage {
+    private static final String DATA_STORAGE_FILE_NAME = "tasks.csv";
+
+    /**
+     * Read the stored data from the application data directory.
+     * @return a list containing the stored tasks.
+     * @see DataStorage#getDataStoragePath
+     */
+    public List<Task> read() {
+        List<Task> tasks = new ArrayList<>();
+        try {
+            File dataFile = new File(getDataStoragePath(), DATA_STORAGE_FILE_NAME);
+            dataFile.getParentFile().mkdirs(); // create the data directory if not exists
+            dataFile.createNewFile(); // create the file if not exists
+            FileInputStream inputStream = new FileInputStream(dataFile);
+            CSVReader reader = new CSVReader(new InputStreamReader(inputStream));
+            for (String[] parts : reader) {
+                Task task;
+                boolean completed = Objects.equals(parts[1], "1");
+                switch (parts[0]) {
+                case "T": {
+                    task = new Todo(completed, parts[2]);
+                    break;
+                }
+                case "E": {
+                    LocalDate time = LocalDate.parse(parts[3]);
+                    task = new Event(completed, parts[2], time);
+                    break;
+                }
+                case "D": {
+                    LocalDate time = LocalDate.parse(parts[3]);
+                    task = new Deadline(completed, parts[2], time);
+                    break;
+                }
+                default: {
+                    throw new CorruptedDataException();
+                }
+                }
+                tasks.add(task);
+            }
+        } catch (SecurityException | IOException err) {
+            System.err.println("Warning: unable to read data storage. "
+                    + "Your task list is not loaded.");
+        } catch (DateTimeException | CorruptedDataException | IndexOutOfBoundsException err) {
+            System.err.println("Warning: Corrupted data storage. "
+                    + "Your task list may not be fully loaded.");
+        }
+        return tasks;
+    }
+
+    /**
+     * Write the task list into the application data directory.
```

Comment:
"Writes" instead of "Write". Use this tense for other JavaDoc comments as well.

Reply:
Fixed. Thanks for reminding!